src/test/java/org/passay/doc/ClassGenerator.java

```diff
@@ -0,0 +1,312 @@
+import java.nio.charset.StandardCharsets;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import com.sun.codemodel.JClassAlreadyExistsException;
+import com.sun.codemodel.JCodeModel;
+import com.sun.codemodel.JDefinedClass;
+import com.sun.codemodel.JDocComment;
+import com.sun.codemodel.JMethod;
+import com.sun.codemodel.JMod;
+
+/**
+ * Utility class for creating Java POJOs from documentation source.
+ *
+ * @author  Middleware Services
+ */
+public class ClassGenerator
+{
+
+  /** Package to create classes in. */
+  private static final String PACKAGE_TO_CREATE = "org.passay.doc";
+
+  /** Packages to import for compilation. */
+  private static final String[] PACKAGES_TO_IMPORT = new String[] {
+    "java.io",
+    "java.util",
+    "org.cryptacular.bean",
+    "org.cryptacular.spec",
+    "org.passay",
+    "org.passay.dictionary",
+    "org.passay.dictionary.sort",
+    "org.passay.entropy",
+  };
+
+  /** String containing all import statements. */
+  private static final String IMPORT_STATEMENTS;
+
+  /** Buffer size for reading and writing files. */
+  private static final int BUFFER_SZIE = 2048;
+
+  /** Code model for java class creation. */
+  private final JCodeModel codeModel = new JCodeModel();
+
+  /** Sections to build beans for. */
+  private final Map<String, List<String>> sections = new HashMap<>();
+
+  /** Initialize {@link #IMPORT_STATEMENTS}. */
+  static {
+    final StringBuilder sb = new StringBuilder();
+    for (String p : PACKAGES_TO_IMPORT) {
+      sb.append("import ").append(p).append(".*;").append("\n");
+    }
+    IMPORT_STATEMENTS = sb.toString();
+  }
+
+
+  /**
+   * Creates a new class generator.
+   *
+   * @param  url  to download zipped source from
+   * @param  path  to write files to
+   *
+   * @throws  IOException  if the source cannot be downloaded, unzipped and read
+   */
+  public ClassGenerator(final String url, final String path)
+    throws IOException
+  {
+    // download the zipped source
+    download(url, path + "/source.zip");
+
+    // unzip the source
+    unzip(path + "/source.zip", path + "/doc-sources");
+
+    // read the source files
+    final Path sourceDir = Paths.get(path + "/doc-sources");
+    Files.walkFileTree(sourceDir, new SimpleFileVisitor<Path>() {
+      @Override
+      public FileVisitResult visitFile(final Path file, final BasicFileAttributes attrs)
+        throws IOException
+      {
+        if (attrs.isRegularFile()) {
+          final String content = new Scanner(file.toFile()).useDelimiter("\\Z").next();
+          final String name = file.getName(file.getNameCount() - 2).toString();
```

Comment:
Section names are arbitrary, they just have to be unique. Since they are read from the zip, uniqueness is guaranteed. In the case of passay, there is a single section named 'reference'. Ldaptive uses multiple sections.